module/Application/src/Application/Service/AbstractAclService.php

```diff
@@ -34,27 +34,69 @@ public function getRole()
         return $this->getServiceManager()->get('user_role');
     }
 
-    /**
-     * Check if a operation is allowed for the current role.
+
+	
+	/**
+     * Check if a operation is allowed for an user.
      *
+	 * If no user is given, this will assume the only role is guest
      * If no resource is given, this will use the resource given by
      * {@link getDefaultResourceId()}.
      *
      * @param string $operation Operation to be checked.
      * @param string|ResourceInterface $resource Resource to be checked
-     *
+	 * @param User $user user to check operation for if null then current user is used     
+	 *
      * @return boolean
      */
-    public function isAllowed($operation, $resource = null)
+    public function isAllowed($operation, $resource=null, $user=null)
     {
+		
         if (null === $resource) {
             $resource = $this->getDefaultResourceId();
         }
+		if (null === $user){
+			$user = $this -> getRole();
+		}
+		
+		
+		$roles = array();
+		if ($user instanceof User ){
+			$roles = $user->getRoleNames();
+		}else{
+			$roles[] = "guest";
+		}
+		
+		var_dump($roles);
+		
+		foreach($roles as $role){
+			if($this->getAcl()->isAllowed($role, $resource, $operation)){
+				return true;
+			}
+		}
+		return false;
+    }
```

Comment:
Dit is dus juist fout. Ten eerste staat er nog een `var_dump` in. Maar vooral, de roles van een user zijn al goed gezet in de ACL, waardoor dit helemaal onnodig is. Zie https://github.com/GEWIS/gewisweb/blob/master/module/User/Module.php#L154